src/QsCompiler/Tests.Compiler/TestCases/LocalVerification.qs

```diff
@@ -9,6 +9,128 @@ namespace Microsoft.Quantum.Testing.LocalVerification {
     open Microsoft.Quantum.Testing.TypeChecking;
 
 
+    // type argument inference
+
+    operation TypeArgumentsInference1<'T>(cnt: Int, arg : 'T) : Unit {
```

Comment:
Could you add some descriptions to these so that it is easier to understand the tested scenario in each?

Reply:
Having a description for each test case is not something I want to start to have. The reason for that is that I want to keep the bar for adding test cases as low as possible. Ideally we would even just autogenerate all combinations that can occur in Q# for certain things. Realistically, what happens if we were to require documentation on test cases is that there will be less test cases. Taking this example here, there are 41 test cases. Asking to have a description for each of those 41 cases is not reasonable. Each case is clearly understandable in the sense that without knowing how the test works, you see the Q# code, and what diagnostics it is supposed to produce. If you have a case in mind that you think is not covered, then I'll happily add it.

Reply:
I don't think it is very unreasonable or burdensome to a developer to have them given descriptive names/descriptions to the code they write. I have a hard time understanding the scenario being tests in these cases just by looking at them, which also makes them difficult to review as part of a PR because I don't know if they correctly represent the intended scenario, or if the scenario is even valid as a test case. As far as producing less test, it is far more useful to have less tests that are understandable than a bunch of tests that developers don't know what to do with if they break. For example, if one of these tests starts failing, it will be especially difficult for me to know how to fix it if I don't understand the test.
What is the purpose of VariableDeclaration14, for example? Or VariableDeclaration21?

I'm not sure I see how generating test scenarios would work, but we can deal with that if we get to it.

Reply:
Interesting example of a Python library for automatically generating test cases: https://hypothesis.readthedocs.io/en/latest/

Reply:
Autogenerating tests is actually a pretty ideal test setup, and while the quality of tests is as if not more important than their number, test quality and coverage are what we optimize for in our dev process when it comes to guidance for writing tests. 

Consider these cases the manual version of what you would get if we were using an automatic test generation tool - they check all combinations of certain patterns in the Q# language and what diagnostics are generated for each one. Each combination is unique, so I am not happy to delete any of the test cases - they are all needed and they all test different things.
Each test case is a statement or expression inside an operation, so what the diagnostics should apply to is I believe fairly clear. Naming them differently will not help identify the issue if any of these cases fails.

Reply:
To be clear, no one was suggesting that any of these cases get deleted. While I disagree with the decision, I can respect that you have made a decision here, and I can appreciate the explanation regarding wanting to make the tests similar to if they were automatically generated.